File: Smart_Mess_Backend/readPdf.js
const fs = require('fs');
const pdfParse = require('pdf-parse');
console.log('pdfParse type:', typeof pdfParse);
if (typeof pdfParse === 'object') {
    console.log('Keys:', Object.keys(pdfParse));
}
const dataBuffer = fs.readFileSync('C:/Users/Mahesh/OneDrive/Desktop/SmartMess/revised mess menu.pdf');

try {
    const fn = typeof pdfParse === 'function' ? pdfParse : (pdfParse.default || pdfParse);
    fn(dataBuffer).then(function(data) {
        console.log(data.text);
    });
} catch(err) {
    console.error(err);
}

File: Smart_Mess_Backend/readPdf.py
import PyPDF2

def read_pdf():
    try:
        reader = PyPDF2.PdfReader('C:/Users/Mahesh/OneDrive/Desktop/SmartMess/revised mess menu.pdf')
        text = ""
        for page in reader.pages:
            text += page.extract_text() + "\n"
        print(text)
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    read_pdf()
